demo/lambdas/worker.rb

```diff
@@ -1,15 +1,32 @@
 require 'aws-sdk-states'
-ENV['AWS_SECRET_ACCESS_KEY'] = 'test'
-ENV['AWS_ACCESS_KEY_ID'] = 'test'
-$endpoint = "http://#{ENV['LOCALSTACK_HOSTNAME']}:4566"
-$state_machine_arn = 'arn:aws:states:us-east-1:000000000000:stateMachine:processingStateMachine'
+
+$localstack_hostname = ENV['LOCALSTACK_HOSTNAME']
+$endpoint = "http://#{$localstack_hostname}:4566"
+if $localstack_hostname
+    ENV['AWS_SECRET_ACCESS_KEY'] = 'test'
+    ENV['AWS_ACCESS_KEY_ID'] = 'test'
+end
 
 def triggerProcessing(event:, context:)
+    if $localstack_hostname
+        client = Aws::States::Client.new(:endpoint => $endpoint)
+    else
+        client = Aws::States::Client.new()
+    end
+
+    # get state machine ARN - TODO consider adding paging
+    result = client.list_state_machines()
+    sm_arn = ""
+    for sm in result.state_machines do
+        if sm.name == 'processingStateMachine'
+            sm_arn = sm.state_machine_arn
+        end
+    end
+
     records = event['Records']
-    client = Aws::States::Client.new(:endpoint => $endpoint)
     for rec in records do
         result = client.start_execution({
-            state_machine_arn: $state_machine_arn,
+            state_machine_arn: sm_arn,
```

Comment:
Will this "blow up" if the 'processingStateMachine' is for whatever reason not present (= no arn will be set)?

Reply:
Yep, good point - it definitely would be blowing up. We could also inject the SFn ARN into the Lambda environment - let me make that change quickly.. 👍